codecs/wp2/enc/wp2_enc.cpp

```diff
@@ -0,0 +1,205 @@
+
+  // Size of tiles (width/height) in pixels. Tiles are always square. Each
+  // tile is compressed independently, possibly in parallel.
+  // Valid values: 64, 128, 256, 512 (see format_constants.h)
+  // 0 means size is chosen automatically.
+  int tile_size = 0;
+
+  // Algorithm for dividing the image into blocks.
+  // PartitionMethod partition_method = MULTIPASS_PARTITIONING;
+  // The set of allowed block sizes for image partitioning.
+  // PartitionSet partition_set = SOME_RECTS;
+  // If true, use binary space partitioning instead of floating partition.
+  bool partition_snapping = true;
+
+  // The set of predictors that can be used for reconstruction.
+  // PredictorSet predictor_set = PredictorSet::VP8_PREDICTORS;
+
+  WP2::Csp csp_type = WP2::Csp::kYCoCg; // Colorspace.
+
+  WP2::EncoderConfig::UVMode uv_mode =
+      WP2::EncoderConfig::UVMode::UVMode420; // Default sub-sampling mode for
+                                             // U/V planes.
+
+  // int preprocessing = 0;           // Preprocessing filter.
+  // int preprocessing_strength = 0;  // Range: [0 .. 100]
+
+  // bool use_random_matrix = false;  // Experimental.
+  // bool store_grain = false;        // Experimental: store grain info
+
+  // // Parameters related to lossless compression only:
+  // bool use_delta_palette = false;  // Reserved for future lossless feature.
+
+  // // Performance parameters (no impact on encoded bytes):
+  // int thread_level = 0;     // If non-zero, try and use multi-threaded
+  // encoding. bool low_memory = false;  // Memory usage reduction (but CPU use
+  // increase).
+
+  // // Neural compression:
+  // int use_neural_compression = 0;       // Neural network compression.
+  // const char* graphdef_path = nullptr;  // Directory holding encoder /
+  // decoder
+  //                                       // graphdefs structure:
+  //                                       // base/qq/[en|de]coder.pbbin
+
+  // EncoderInfo* info = nullptr;   // If not
+};
+
+val encode(std::string image_in, int image_width, int image_height,
+           WP2Options options) {
+  WP2Status status;
+  uint8_t *image_buffer = (uint8_t *)image_in.c_str();
+  WP2::ArgbBuffer src = WP2::ArgbBuffer();
+  status = src.Import(WP2_rgbA_32, // Format. WP2_RGBA_32 is the same but NOT
+                                   // premultiplied alpha
+                      image_width, image_height, image_buffer, 4 * image_width);
+  if (status != WP2_STATUS_OK) {
+    return val(1);
+  }
+
+  WP2::MemoryWriter memory_writer;
+  WP2::EncoderConfig config;
+  config.quality = options.quality;
+  config.alpha_quality = options.alpha_quality;
+  config.speed = options.speed;
+  config.pass = options.pass;
+  config.sns = options.sns;
+  status = WP2::Encode(src, &memory_writer, config);
+  if (status != WP2_STATUS_OK) {
+    return val(2);
+  }
+
+  return val(typed_memory_view(memory_writer.size_, memory_writer.mem_));
+  // Lol I forgot to add the free here.
```

Comment:
Judging by the comment, you already know it, but also needs upgrade to return an owned Uint8Array :)